test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -178,19 +180,33 @@ func testVolumeUnmountsFromDeletedPod(c clientset.Interface, f *framework.Framew
 	nodeIP = nodeIP + ":22"
 
 	By("Expecting the volume mount to be found.")
-	result, err := framework.SSH(fmt.Sprintf("mount| grep %s", string(clientPod.UID)), nodeIP, framework.TestContext.Provider)
-	Expect(err).NotTo(HaveOccurred())
-	Expect(result.Code).To(BeZero())
+	result, err := framework.SSH("mount | grep "+string(clientPod.UID), nodeIP, framework.TestContext.Provider)
+	framework.LogSSHResult(result)
+	Expect(err).NotTo(HaveOccurred(), "Encountered SSH error.")
+	Expect(result.Code).To(BeZero(), "Expected grep exit code of 0, got "+string(result.Code))
 
-	By("Restarting the kubelet.")
+	By("Stopping the kubelet.")
 	kubeletCommand(kStop, c, clientPod)
-	framework.ExpectNoError(framework.DeletePodWithWait(f, c, clientPod), "Failed to delete pod ", clientPod.Name)
+	defer func() {
+		if err != nil {
+			kubeletCommand(kStart, c, clientPod)
```

Comment:
In this instance, I think it's okay.  Issuing `start` on a running kubelet will make it restart.  This defer is primarily to protect against the kubelet being left stopped if an error happens while deleting the pod or waiting for the pod to terminate.  In testing, timeouts here left the kubelet stopped after the test finished.

The side effect being if an error occurs after the kubelet is started (SSH fails or the volume is still attached), then it will be restarted once before the function returns.  `kubeletCommand` waits for the node to come back online before returning, so this will add a few seconds to run time.